### DISCUSSION TITLE
Functions for calculating the width in advance

### DISCUSSION THREAD

**Discussion by 0lru** [ocornut/imgui#4114] [Old/Archived discussions]
Hi. I'm trying to create a decent layout system among other things: 
https://github.com/0lru/oogui. Please have a look at the second example of the Readme.md.

However, in CSS we have: `flex-basis: auto`. This would require calculating the with
of any widget in advance as well as the height. Looking at the code of ImGUI this should
be doable. 

Would it be possible to add? Maybe something like this (pseudocode):
```cpp
DoNotRenderTheVeryNextItem() // <- new command
ImGui::Button...
```
Looking at the code, the rest of it is already implemented, since one can get the bounds
of the last item created.

**Comment by 0lru** [comment]
Example/Use Case:
On the left, you can see a flexible, vertical layout implementation. The content is centered horizontally.
The width calculation for CheckBox and InputText isn't correct here (I think it could/should be supported by ImGui natively).
![flexible](https://user-images.githubusercontent.com/61606260/124159101-098db500-da9b-11eb-88e8-4fb142b992f7.PNG)

**Reply by rokups** [reply]
There are no such functions at the moment. What we do is manually calculate widget sizes in advance when this is needed. However this is a very fragile process that may break should internal widget layout change (even if that is unlikely). This also does not scale as there is no way to support arbitrary user widgets automatically.

@ocornut might have a better input on the matter.

**Reply by ocornut** [reply]
Not answering on github Discussions sorry.
